src/openfermion/hamiltonians/_bose_hubbard.py

```diff
@@ -0,0 +1,133 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""This module constructs Hamiltonians for the Bose-Hubbard model."""
+from __future__ import absolute_import
+
+from openfermion.ops import BosonOperator
+from openfermion.utils import (hermitian_conjugated, boson_number_operator,
+                               up_index, down_index)
+
+
+def bose_hubbard(x_dimension, y_dimension, tunneling, interaction,
```

Comment:
I agree - potentially the bose_hubbard and fermi_hubbard would also differ in their signatures, and the terminology used, so it might make sense to have an underlying _hubbard function, which is then called by a minimal top-level bose_hubbard and fermi_hubbard functions.